gtk/src/toga_gtk/libs/utils.py

```diff
@@ -11,3 +14,36 @@ def gtk_text_align(alignment):
         CENTER: (0.5, Gtk.Justification.CENTER),
         JUSTIFY: (0.0, Gtk.Justification.FILL),
     }[alignment]
+
+
+if GTK_VERSION >= (4, 0, 0):  # pragma: no-cover-if-gtk3
+
+    def create_toga_native(native_gtk_class):  # pragma: no-cover-if-gtk3
+        """Create a new native class from a native gtk class, whose virtual functions
+        could be safely overridden."""
+        toga_native_class = type(
+            native_gtk_class.__gtype__.name,
+            (native_gtk_class,),
+            {"base_class": native_gtk_class},  # Store the base class type
+        )
+        return toga_native_class
+
+    class WeakrefCallable:  # pragma: no-cover-if-gtk3
```

Comment:
If this is the same WeakrefCallable implementation that is used by Winforms, then we should pull it out as a common resource in core, not duplicate that logic.